### ISSUE TITLE
flycheck-rtags not showing errors on files with irregular extensions such as `.tpp` 

### ISSUE THREAD

**Issue by Deanseo** [Andersbakken/rtags#1330]
**Describe the bug**
`flycheck-rtags` works fine, correctly showing errors on buffers except for files whose extensions are uncommon ones such as `.ipp`, `.tpp`, or `.h.misc`.

**To Reproduce**
1. Create a normal project, and make sure files are indexed correctly.
2. Make an intentional syntax error on any file whose extension is not typical `.h`, `.hpp`, `.cpp` or etc.
3. See whether or not the errors are indeed printed on the buffer.

**Expected behavior**
Syntax checks should be performed regardless of the extension of the files

**Environment (please complete the following information):**
 - OS/Version: MacOS 10.14
 - RTags(rdm,rc): 2.31.128
 - Emacs: 26.2
 - LLVM/Clang: Apple LLVM version 10.0.1 (clang-1001.0.46.4)

**Additional context**
I use `cmake-ide` to let most configurations auto-configured, 
The following lines are part of my `init.el`:

```
(use-package rtags
    :ensure t
    :init
    (defun my-rtags-mode-hook ()
      (local-unset-key (kbd "C-c C-r"))
      (local-set-key (kbd "C-c C-r") #'rtags-rename-symbol)
      
      (local-unset-key (kbd "M-."))
      (local-set-key (kbd "M-.") #'rtags-find-symbol-at-point)

      (local-unset-key (kbd "M->"))
      (local-set-key (kbd "M->") #'rtags-find-symbol))

    (add-hook 'c++-mode-hook #'my-rtags-mode-hook)
    (add-hook 'c-mode-hook #'my-rtags-mode-hook)

    :config
    (setq rtags-find-file-case-insensitive t)
    (setq rtags-symbolnames-case-insensitive t)
    (setq rtags-autostart-diagnostics t)
    (setq rtags-timeout 1100)
    (setq rtags-completions-enabled t)
    (rtags-set-periodic-reparse-timeout 1.1))

(use-package flycheck-rtags
    :ensure t
    :after rtags
    :init
    (defun my-flycheck-rtags-setup ()
      (flycheck-select-checker 'rtags)
      (setq-local flycheck-highlighting-mode nil)
      (setq-local flycheck-check-syntax-automatically nil)
      (setq-local flycheck-clang-standard-library "c++17")

      (flycheck-mode t))
    
    :config
    (add-hook 'c++-mode-hook #'my-flycheck-rtags-setup)
    (add-hook 'c-mode-hook #'my-flycheck-rtags-setup))

(use-package cmake-ide
    :ensure t
    :init
    ;; ... just some key-bindings
    :config
    (cmake-ide-setup))
```

**Comment by Deanseo** [comment]
To elaborate a little more on this,

Suppose I would like to use `.tpp` to split a header file with templates to a header and implementation section, as follows:

```
// Foo.hpp
class Foo
{
public:
     template <class T>
     void Bar(const T&);
};

#include "Foo.tpp"

// Foo.tpp
template <class T>
void Foo::Bar(const T&) { ... }
```

In this case, I cannot _have flycheck fly_.

Now, what's interesting is that if I rename `Foo.tpp` with `Foo.cpp`, `flycheck-rtags` suddenly does work. 

It gets turned off only when the file extension is unrecognized.

So I'm guessing there should be a way to tell it to recognize some specific extensions...?

**Comment by cslux** [comment]
@Deanseo In what mode major mode is that file opened? For me it was opened in `fundamental-mode`. Does it work when you switch to `c++-mode`?

**Comment by Deanseo** [comment]
@cslux `C-h m` in a `.tpp` file shows **C++//l mode defined in ‘cc-mode.el’:** is being used, which is the exactly same mode as the mode used in other regular files (`.cpp` or `.h`).

For minor modes, `(describe-mode)` in `cpp` files show the following information:
```
Enabled minor modes: Abbrev Async-Bytecomp-Package Auto-Composition
Auto-Compression Auto-Encryption Auto-Revert Back-Button Blink-Cursor
Company Delete-Selection Desktop-Save Diff-Auto-Refine Electric-Indent
Eyebrowse File-Name-Shadow Flycheck Flyspell Font-Lock Global-Company
Global-Font-Lock Global-Git-Commit Global-Linum Global-Magit-File Helm
Highlight-Indent-Guides Irony Line-Number Linum Magit-Auto-Revert
Magit-File Menu-Bar Mouse-Wheel Override-Global Recentf Shell-Dirtrack
Size-Indication Tool-Bar Tooltip Transient-Mark

(Information about these minor modes follows the major mode info.)

C++//l mode defined in ‘cc-mode.el’:
Major mode for editing C++ code.
To submit a problem report, enter ‘M-x c-submit-bug-report’ from a
c++-mode buffer.  This automatically sets up a mail buffer with
version information already added.  You just need to add a description
of the problem, including a reproducible test case, and send the
message.
```

in `.tpp` files, also show the following information which is pretty much the same, except that it's missing the **Auto-Revert** mode:
```
Enabled minor modes: Abbrev Async-Bytecomp-Package Auto-Composition
Auto-Compression Auto-Encryption Back-Button Blink-Cursor Company
Delete-Selection Desktop-Save Diff-Auto-Refine Electric-Indent
Eyebrowse File-Name-Shadow Flycheck Flyspell Font-Lock Global-Company
Global-Font-Lock Global-Git-Commit Global-Linum Global-Magit-File Helm
Highlight-Indent-Guides Irony Line-Number Linum Magit-Auto-Revert
Magit-File Menu-Bar Mouse-Wheel Override-Global Recentf Shell-Dirtrack
Size-Indication Tool-Bar Tooltip Transient-Mark

(Information about these minor modes follows the major mode info.)

C++//l mode defined in ‘cc-mode.el’:
Major mode for editing C++ code.
To submit a problem report, enter ‘M-x c-submit-bug-report’ from a
c++-mode buffer.  This automatically sets up a mail buffer with
version information already added.  You just need to add a description
of the problem, including a reproducible test case, and send the
message.
```

**Comment by Andersbakken** [comment]
Sorry about the long delay. I haven't been monitoring things lately. I believe my last push might take care of this. Please reopen if it's still an issue.

**Comment by Deanseo** [comment]
I've updated `rtags` to *2.33.128* and the issue still remains.
`flycheck-rtags` only detects compile errors at regular extensions such as `.cpp`.

> Sorry about the long delay. I haven't been monitoring things lately. 
Totally understandable. :)
Appreciate your great work always! 👍

**Comment by Deanseo** [comment]
I tried `.tpp` that's newly added at [this commit](https://github.com/Andersbakken/rtags/commit/dcd3747ce168cbb8a885f8f3db95bd304e8e3ee6), I've updated `rtags` to *2.33.128* and the issue still remains.

`flycheck-rtags` only detects compile errors at regular extensions such as `.cpp`.

Is this issue somewhat related to #1340 ?

> Sorry about the long delay. I haven't been monitoring things lately. 
Totally understandable. :)
Appreciate your great work always! 👍
